include/libtorrent/announce_entry.hpp

```diff
@@ -142,6 +126,51 @@ namespace libtorrent {
 
 TORRENT_VERSION_NAMESPACE_2
 
+	// announces are sent to each tracker using every listen socket
+	// this class holds information about one listen socket for one tracker
+	struct TORRENT_EXPORT announce_endpoint
+	{
+		friend class lt::torrent;
+		friend struct announce_entry;
+
+		// internal
+		explicit announce_endpoint(aux::listen_socket_handle const& s);
+
+		// the local endpoint of the listen interface associated with this endpoint
+		tcp::endpoint local_endpoint;
+
+		// torrents can be announced using multiple info hashes
+		// for different protocol versions
+
+		// info_hashes[0] is the v1 info hash (SHA1)
+		// info_hashes[1] is the v2 info hash (truncated SHA-256)
+		announce_infohash info_hashes[2];
```

Comment:
I wonder if it would make sense to make this `aux::array<announce_infohash, 2, protocol_version> info_hashes` instead. It might make it clearer which one is which version.

Reply:
I'll test this out